From ad502606fcb55be1ddc03d9e2fa4d7cc3b367628 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 25 Jan 2016 03:08:33 +0700
Subject: [PATCH 58/64] Revert "move torch to vanir activity and use the
 correct drawable"

This reverts commit f26d705a96e298be4848fcc56d7aa82ab2ddcd18.

Change-Id: I54b599c64d65d5edc71ea5024b1fb2d780009852
---
 core/java/com/android/internal/util/aokp/AwesomeAction.java        | 6 +++++-
 core/java/com/android/internal/util/aokp/AwesomeConstants.java     | 7 +++++++
 core/java/com/android/internal/util/aokp/NavBarHelpers.java        | 3 +--
 packages/SystemUI/AndroidManifest.xml                              | 2 +-
 .../SystemUI/src/com/android/systemui/AOKPSearchPanelView.java     | 1 +
 packages/SystemUI/src/com/android/systemui/CreateShortcut.java     | 2 +-
 6 files changed, 16 insertions(+), 5 deletions(-)

diff --git a/core/java/com/android/internal/util/aokp/AwesomeAction.java b/core/java/com/android/internal/util/aokp/AwesomeAction.java
index a68f420..10214ff 100755
--- a/core/java/com/android/internal/util/aokp/AwesomeAction.java
+++ b/core/java/com/android/internal/util/aokp/AwesomeAction.java
@@ -55,6 +55,7 @@ import com.android.internal.statusbar.IStatusBarService;
 import java.net.URISyntaxException;
 import java.util.List;
 
+import com.android.internal.util.cm.TorchConstants;
 import static com.android.internal.util.aokp.AwesomeConstants.AwesomeConstant;
 import static com.android.internal.util.aokp.AwesomeConstants.fromString;
 import com.vanir.util.TaskUtils;
@@ -122,7 +123,10 @@ public class AwesomeAction {
                 case ACTION_POWER:
                     triggerVirtualKeypress(KeyEvent.KEYCODE_POWER, STANDARD_FLAGS);
                     break;
-
+                case ACTION_TORCH:
+                    Intent intentTorch = new Intent(TorchConstants.ACTION_TOGGLE_STATE);
+                    mContext.sendBroadcast(intentTorch);
+                    break;
                 case ACTION_LAST_APP:
                     TaskUtils.toggleLastApp(mContext);
                     break;
diff --git a/core/java/com/android/internal/util/aokp/AwesomeConstants.java b/core/java/com/android/internal/util/aokp/AwesomeConstants.java
index c565eb5..38658fb 100644
--- a/core/java/com/android/internal/util/aokp/AwesomeConstants.java
+++ b/core/java/com/android/internal/util/aokp/AwesomeConstants.java
@@ -57,6 +57,7 @@ public class AwesomeConstants {
         ACTION_NOTIFICATIONS { @Override public String value() { return "**notifications**";}},
         ACTION_VOICEASSIST   { @Override public String value() { return "**voiceassist**";}},
         ACTION_LAST_APP      { @Override public String value() { return "**lastapp**";}},
+        ACTION_TORCH         { @Override public String value() { return "**torch**";}},
         ACTION_IME           { @Override public String value() { return "**ime**";}},
         ACTION_KILL          { @Override public String value() { return "**kill**";}},
         ACTION_BLANK         { @Override public String value() { return "**blank**";}},
@@ -176,6 +177,9 @@ public class AwesomeConstants {
             case ACTION_VOICEASSIST:
                 value = res.getString(com.android.internal.R.string.action_voiceassist);
                 break;
+            case ACTION_TORCH:
+                value = res.getString(com.android.internal.R.string.action_torch);
+                break;
             case ACTION_SILENT:
                 value = res.getString(com.android.internal.R.string.action_silent);
                 break;
@@ -262,6 +266,9 @@ public class AwesomeConstants {
             case ACTION_VOICEASSIST:
                 value = getSystemUIDrawable(context, "com.android.systemui:drawable/ic_sysbar_voiceassist");
                 break;
+            case ACTION_TORCH:
+                value = getSystemUIDrawable(context, "com.android.systemui:drawable/ic_sysbar_torch");
+                break;
             case ACTION_SILENT:
                 value = getSystemUIDrawable(context, "com.android.systemui:drawable/ic_sysbar_silent");
                 break;
diff --git a/core/java/com/android/internal/util/aokp/NavBarHelpers.java b/core/java/com/android/internal/util/aokp/NavBarHelpers.java
index f92e86d..e67af86 100644
--- a/core/java/com/android/internal/util/aokp/NavBarHelpers.java
+++ b/core/java/com/android/internal/util/aokp/NavBarHelpers.java
@@ -43,8 +43,7 @@ public class NavBarHelpers {
             AwesomeConstant.ACTION_ARROW_UP,
             AwesomeConstant.ACTION_ARROW_DOWN,
             /* these are just not implemented yet: */
-            AwesomeConstant.ACTION_WIDGETS,
-            AwesomeConstant.ACTION_APP_WINDOW
+            AwesomeConstant.ACTION_WIDGETS
     };
 
     private NavBarHelpers() {
diff --git a/packages/SystemUI/AndroidManifest.xml b/packages/SystemUI/AndroidManifest.xml
index 42698a4..cd49b00 100755
--- a/packages/SystemUI/AndroidManifest.xml
+++ b/packages/SystemUI/AndroidManifest.xml
@@ -365,7 +365,7 @@
         <activity
             android:name="com.android.systemui.vanir.Torch"
             android:label="@string/toggle_torch"
-            android:icon="@drawable/ic_sysbar_torch"
+            android:icon="@drawable/ic_qs_torch_on"
             android:theme="@android:style/Theme.Holo.Panel"
             android:excludeFromRecents="true"
             android:launchMode="singleTop" >
diff --git a/packages/SystemUI/src/com/android/systemui/AOKPSearchPanelView.java b/packages/SystemUI/src/com/android/systemui/AOKPSearchPanelView.java
index 0ae5085..6aba2df 100644
--- a/packages/SystemUI/src/com/android/systemui/AOKPSearchPanelView.java
+++ b/packages/SystemUI/src/com/android/systemui/AOKPSearchPanelView.java
@@ -275,6 +275,7 @@ public class AOKPSearchPanelView extends FrameLayout implements
         if (action.equals(AwesomeConstant.ACTION_SILENT_VIB.value()) ||
             action.equals(AwesomeConstant.ACTION_VIB.value()) ||
             action.equals(AwesomeConstant.ACTION_POWER.value()) ||
+            action.equals(AwesomeConstant.ACTION_TORCH.value()) ||
             action.equals(AwesomeConstant.ACTION_NOTIFICATIONS.value()) ||
             action.equals(AwesomeConstant.ACTION_SILENT.value())) {
             return false;
diff --git a/packages/SystemUI/src/com/android/systemui/CreateShortcut.java b/packages/SystemUI/src/com/android/systemui/CreateShortcut.java
index a41a3d7..53a6306 100644
--- a/packages/SystemUI/src/com/android/systemui/CreateShortcut.java
+++ b/packages/SystemUI/src/com/android/systemui/CreateShortcut.java
@@ -57,7 +57,7 @@ public class CreateShortcut extends LauncherActivity {
         String c = className.substring(className.lastIndexOf(".") + 1);
 
         if (c.equals("Torch")) {
-            return R.drawable.ic_sysbar_torch;
+            return R.drawable.ic_qs_torch_on;
         } else if (c.equals ("ScreenShotAction")) {
             return R.drawable.ic_sysbar_screenshot;
         } else if (c.equals ("FlushMemory")) {
-- 
2.5.0

